### PR TITLE
Improve keyboard performance revisited

### PR DESCRIPTION
**Pull request by shamilovtim**
This PR incorporates #1171, #1326  by adding the `isKeyboardInternallyHandled` prop. When set to true, a KeyboardAvoidingView is added from react-native in order to make the animation smoother on the textbox above the keyboard.

When the prop is set to false, there is only a View with no animation. This gives the option for users to use their own keyboard handlers.

Also to address @llamaluvr concerns about offering an offset prop for people's chrome. This  prop was never needed because the keyboard avoider only smooths the animation, it doesn't actually handle the keyboard.

Let me know if you have any questions.

### CODE REVIEW

**Comment by xcarpentier** [comment]
Can you see #1524?

**Comment by shamilovtim** [comment]
I pulled #1524 into this branch but my feeling is that it doesn't work correctly. I left some comments in there

File: src/GiftedChat.tsx [hunk 1]
```diff
@@ -632,22 +612,41 @@ class GiftedChat<TMessage extends IMessage = IMessage> extends React.Component<
...
-      >
-        <MessageContainer
-          {...this.props}
-          invertibleScrollViewProps={this.invertibleScrollViewProps}
-          messages={this.getMessages()}
-          forwardRef={this._messageContainerRef}
-        />
-        {this.renderChatFooter()}
-      </AnimatedView>
+      <>
```
[the remark this answers is not in the document]
Comment (shamilovtim) [reply src/GiftedChat.tsx]: Done

**Comment by shamilovtim** [comment]
To contradict my last comment I am not committing #1524 here so this issues stays separate because it's ready to go

**Comment by xcarpentier** [comment]
Thanks!

**Pull request merged**